From ec623715d2716db078a0248925506d4d75d7fd57 Mon Sep 17 00:00:00 2001
Message-ID: <ec623715d2716db078a0248925506d4d75d7fd57.1752837271.git.jdenemar@redhat.com>
From: Peter Krempa <pkrempa@redhat.com>
Date: Thu, 3 Jul 2025 14:18:46 +0200
Subject: [PATCH] virDomainDriverAutoShutdownDoSave: Don't attempt to save
 transient VMs

Commit 84bb136c31e added code that intended to skip the save of
transient domains but did so only in the setup part where we pause the
VMS. The second loop that actually attempts to save the VM was not
modified so we'd still try saving them:

 Jul 03 14:15:13 andariel virtqemud[247210]: auto-shutdown: unable to perform managed save of 'cd3': Requested operation is not valid: cannot do managed save for transient domain

Fixes: 84bb136c31e
Signed-off-by: Peter Krempa <pkrempa@redhat.com>
Reviewed-by: Pavel Hrdina <phrdina@redhat.com>
(cherry picked from commit e1f84ca87e20c64999d3e6e1daccbd86dda743ef)

https://issues.redhat.com/browse/RHEL-95196
---
 src/hypervisor/domain_driver.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/hypervisor/domain_driver.c b/src/hypervisor/domain_driver.c
index cce6c64d1b..d8ccee40d5 100644
--- a/src/hypervisor/domain_driver.c
+++ b/src/hypervisor/domain_driver.c
@@ -779,6 +779,10 @@ virDomainDriverAutoShutdownDoSave(virDomainPtr *domains,
     }
 
     for (i = 0; i < numDomains; i++) {
+        if ((transient[i] && cfg->trySave == VIR_DOMAIN_DRIVER_AUTO_SHUTDOWN_SCOPE_PERSISTENT) ||
+            (!transient[i] && cfg->trySave == VIR_DOMAIN_DRIVER_AUTO_SHUTDOWN_SCOPE_TRANSIENT))
+            continue;
+
         virSystemdNotifyStatus("Saving '%s' (%zu of %zu)",
                                virDomainGetName(domains[i]), i + 1, numDomains);
         VIR_INFO("Saving '%s'", virDomainGetName(domains[i]));
-- 
2.50.1
